tests/PHPStan/Type/TypeCombinatorTest.php

```diff
@@ -3934,6 +3975,46 @@ public function dataIntersect(): iterable
 				IntersectionType::class,
 				'literal-string&lowercase-string',
 			],
+			[
+				[
+					new StringType(),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				IntersectionType::class,
+				'uppercase-string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryNumericStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				IntersectionType::class,
+				'numeric-string&uppercase-string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryNonFalsyStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				IntersectionType::class,
+				'non-falsy-string&uppercase-string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryNonEmptyStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				IntersectionType::class,
+				'non-empty-string&uppercase-string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryLiteralStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				IntersectionType::class,
+				'literal-string&uppercase-string',
```

Comment:
You should add a test for `lowercase-string&uppercase-string` which should stay as if.

Reply:
Added in 96773a4d2